Kernel/Arch/aarch64/kprintf.cpp

```diff
@@ -64,3 +64,26 @@ void kernelearlyputstr(char const* characters, size_t length)
     for (size_t i = 0; i < length; ++i)
         console_out(characters[i]);
 }
+
+void dbgputstr(char const* characters, size_t length)
+{
+    if (!characters)
+        return;
+
+    auto& uart = Kernel::RPi::UART::the();
+    uart.print_str(characters, length);
+
+    for (size_t i = 0; i < length; ++i)
+        console_out(characters[i]);
+}
+
+void dbgputstr(StringView view)
+{
+    dbgputstr(view.characters_without_null_termination(), view.length());
+}
+
+void dbgputchar(char ch)
+{
+    auto& uart = Kernel::RPi::UART::the();
+    uart.send(ch);
```

Comment:
dbgputstr goes to uart and console, should this one too?

Reply:
I'm really not sure, I'm not familiar with the console code in the kernel. For now, I've implemented `dbgputchar` in terms of `kernelputstr`, so it will also send the characters to the console. We can always fix this later if it turns out we shouldn't be sending the character to the console, or if someone merges this code with `Kernel/kprintf.cpp`.